### DISCUSSION TITLE
Nerd Fonts 3.2.0 / 3.2.1 Release

### DISCUSSION THREAD

**Discussion by Finii** [ryanoasis/nerd-fonts#1570] [General]
Here are considerations revolving around the upcoming 3.2.0 release.

Issues down to 11 ;-)

![2024-04-04_12-53](https://github.com/ryanoasis/nerd-fonts/assets/16012374/926d06a7-eb60-4680-b9ac-45264829e108)

**Comment by Finii** [comment]
## Font updates

| Original Font Name and Repository      | ver        | update  |
|:---------------------------------------|:-----------|:--------|
| 0xProto                   | 1.602      | 1.603  |
| IBM Plex Mono            | 2.3        | 2.004  | 
| Commit Mono                | 1.141      | 1.143  | 
| Geist Mono             | 1.0.1      | 1.3.0|   
| Inconsolata LGC    | 1.3        | 1.5.2 |  
| Lilex                       | 2.300      | 2.400 |  
| Victor Mono                | 1.5.4      | 1.5.6 |  

**DONE**

## Bugs

## Add glyph to core set

Just Ada stuck.

**DONE**

## New fonts

Need to clean up font suggestion list, based on list in 3.1.0 discussion
Not many new suggestions

Lets see what we have here (master list [here](https://github.com/ryanoasis/nerd-fonts/discussions/1095#discussioncomment-4821624))

| Name | Link | arbitrary rank | arbitrary points | comment | issue |
|---|---|---|---|---|---|
| Courier Prime Code  |  https://quoteunquoteapps.com/courierprime/   | 9 | 13 | | #314
| B612Mono            |  https://b612-font.com/ | 7 | 2 | | #383
| CamingoCode         |  https://janfromm.de/typefaces/camingomono/camingocode/ | 7 | 0| | #386
| Dina Font           |  http://www.dcmembers.com/jibsen/download/61/ | - | 5 | no outline font | #399
| Fira Code iScript   | https://github.com/kencrocken/FiraCodeiScript | 8 | 11| no license, 2 fonts combined | #548
| Sarasa Gothic | https://github.com/be5invis/Sarasa-Gothic | 6 | 5| CJK font | #587
| Recursive | https://github.com/arrowtype/recursive  | 5 | 8| | #845
| Photonico | https://github.com/Photonico/Photonico_Code | 8 | 0| | #995
| Julia Mono | https://juliamono.netlify.app/ | 9 | 0 | | #1011
| DM Mono | https://github.com/googlefonts/dm-mono | 7 | 4| | #1299
| Hybrid Mono | https://github.com/ghazyami/hybrid-mono-fonts | 9 | 0| | #1335
| Twilio Sans Mono | https://github.com/twilio/twilio-sans-mono | 7 | 3 | PR #1465 | #1458
| Iosevka Extended | | 8 | 1 | |#1475
| Titillium | | 9 | 0 | |#1512
| Koda Mono | https://fontesk.com/kode-mono-typeface/ | 9 | 0 | |#1533
| Zed Mono | https://github.com/zed-industries/zed-fonts | 5 | 6 | PR #1504 | - |

_rank: less is better, points: more is better_

Lets see if we can include all up to rank 4, maybe 5 into 3.2.0
* Recursive
* Zed Mono
* (Courier Prime? Why is the rank so low (I mean high)? :thinking: )
* Ubuntu Sans

**DONE**

## New glyphs sets

Check out file icons and feather icons
Nothing this time.

**Comment by Finii** [comment]
* Local test release started... `# [Nerd Fonts]  Total source fonts found: 761`
* A lot stuff broken :unamused: 
  * Add missing Zed mono variant
  * Fix naming problem for Blex (regression)
  * Fix naming problem for Lilex (new non-standard weight)
  * Remove obsolete partched MPlus fonts
  * Fix Terminus-Bold-Italic parsing
  * Fix naming glitch with Semi becoming a token (it's only a modifier)
  * Change RecMonoSemiCasual re-naming
  * Naming problems with updated Noto and `keep_regular`
  * #1586
* Adjust `package.json`
* Trigger `Release` workflow, manually
* Create `changelog.md` manually from `git diff --oneline` and also put onto gh-pages as new release
* Edit release draft comment with changelog
* Update sankey diagram and commit to repo and gh-pages
* Fix wrong tag names for new Poweline glyphs
* Create font previews
* Prepare casks https://github.com/ryanoasis/nerd-fonts/discussions/1074#discussioncomment-5770142
  * Sync fork of https://github.com/Finii/homebrew-cask-fonts
  * Update PAT in _Github_Profile - Settings - Developer_Settings - Personal_Access_Tokens - Fine-grained_Tokens_
  * Put it into the file
* Publish Release
* Trigger Cask workflow manually
* Create PR is cask fork

**Reply by Finii** [reply]
Hmm, these workflows were not triggered, manually doing that now.
Thank you @landfillbaby 

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/d5e603ea-6314-474f-9b09-e156019718da)

**Reply by landfillbaby** [reply]
remember to update the FontPatcher.zip release asset!

**Reply by Finii** [reply]
:1st_place_medal: 

Currently I'm fighting with firefox, it drives my crazy. On one machine there is no keyboard repeat, and emoticon selection is broken etc, while on another machine everything is ok. Firefox is synced. Also tried snap (normally I use debian packages), to no avail. That's the reason I'm a bit distracted, it is as awkward as being forced to use EMACS if you never did before ;-)

**Reply by Finii** [reply]
It's a bug...

https://bugzilla.mozilla.org/show_bug.cgi?id=1880458

switched to `firefox-nighty` :-D

**Reply by Finii** [reply]
## Billing

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/c8395945-c44d-4a65-bb46-845ef9d391b9)

Noto is **by far** the worst font, with 2 hours 21 minutes.

**Reply by Finii** [reply]
`cat bill | tr '\n' '\r' | sed -e 's!\r\([^\t]\)!\n\1!g;s/\r/ /g' | sort -n -k 3`

Not perfect, but the needed manual changes are ok.

| Font              | CI minutes | #files |
|-------------------|---------|--------|
| 0xProto               |   4 |    1   |
| Agave                 |   4 |    2   |
| AurulentSansMono      |   4 |    1   |
| DaddyTimeMono         |   4 |    1   |
| DroidSansMono         |   4 |    1   |
| HeavyData             |   4 |    1   |
| Inconsolata           |   4 |    2   |
| InconsolataGo         |   4 |    2   |
| NerdFontsSymbolsOnly  |   4 |    1   |
| ShareTechMono         |   4 |    1   |
| 3270                  |   5 |    3   |
| Arimo                 |   5 |    4   |
| ComicShannsMono       |   5 |    2   |
| Lekton                |   5 |    3   |
| ProFont               |   5 |    2   |
| ProggyClean           |   5 |    3   |
| Tinos                 |   5 |    4   |
| AnonymousPro          |   6 |    4   |
| BitstreamVeraSansMono |   6 |    4   |
| CodeNewRoman          |   6 |    3   |
| D2Coding              |   6 |    2   |
| DejaVuSansMono        |   6 |    4   |
| EnvyCodeR             |   6 |    3   |
| FantasqueSansMono     |   6 |    4   |
| FiraMono              |   6 |    3   |
| Gohu                  |   6 |    4   |
| Go-Mono               |   6 |    4   |
| Hack                  |   6 |    4   |
| InconsolataLGC        |   6 |    4   |
| Monofur               |   6 |    3   |
| Monoid                |   6 |    4   |
| Mononoki              |   6 |    4   |
| SpaceMono             |   6 |    4   |
| Terminus              |   6 |    4   |
| Cousine               |   7 |    4   |
| UbuntuMono            |   7 |    4   |
| FiraCode              |   8 |    6   |
| Lilex                 |   8 |    6   |
| MartianMono           |   9 |    6   |
| Ubuntu                |   9 |    9   |
| CommitMono            |  10 |    4   |
| Hermit                |  10 |    6   |
| IntelOneMono          |  10 |    8   |
| OpenDyslexic          |  10 |    9   |
| iA-Writer             |  11 |   12   |
| LiberationMono        |  12 |   12   |
| CascadiaMono          |  14 |   12   |
| GeistMono             |  14 |    9   |
| BigBlueTerminal       |  15 |    2   |
| CascadiaCode          |  15 |   12   |
| RobotoMono            |  15 |   12   |
| SourceCodePro         |  16 |   14   |
| UbuntuSans            |  17 |   16   |
| IBMPlexMono           |  18 |   16   |
| Hasklig               |  19 |   14   |
| Overpass              |  19 |   20   |
| Recursive             |  19 |   16   |
| VictorMono            |  24 |   21   |
| Meslo                 |  27 |   24   |
| IosevkaTermSlab       |  29 |   15   |
| JetBrainsMono         |  35 |   32   |
| MPlus                 |  38 |   39   |
| Iosevka               |  51 |   27   |
| Monaspace             |  51 |   40   |
| IosevkaTerm           |  52 |   27   |
| ZedMono               |  60 |   30   |
| Noto                  |   141  |  181   |

**Comment by Finii** [comment]
## Changelog for 3.2.1

This bugfix release is mainly because of the Fira Code progress indicator glyphs. These are used by some applications, and usually could be used via fontfallback from any font (because the codepoints were un-occupied in the Nerd Fonts). But with adding more FontAwesome glyphs we utilized that codepoints for the new glyphs, breaking this common setup.
To fix that we keep these codepoints (there are 9 glyphs) free and un-occupied. So the usualy fallback scheme will continue to work.

Please see below for a list of codepoint that change with this release (which were new in 3.2.0), so I count that not really as breaking, because they were just some days ago (accidentally) introduced.

| name | v3.2.0 codepoint | v3.2.1 codepoint |
|---|---|---|
| fa_less_than              | `EE00` | `EFC3` |
| fa_less_than_equal        | `EE01` | `EFC4` |
| fa_memory                 | `EE02` | `EFC5` |
| fa_microphone_lines_slash | `EE03` | `EFC6` |
| fa_money_bill_wave        | `EE04` | `EFC7` |
| fa_money_bill_1_wave      | `EE05` | `EFC8` |
| fa_money_check            | `EE06` | `EFC9` |
| fa_money_check_dollar     | `EE07` | `EFCA` |
| fa_not_equal              | `EE08` | `EFCB` |
| fa_palette                | `EE09` | `EFCC` |
| fa_square_parking         | `EE0A` | `EFCD` |
| fa_diagram_project        | `EE0B` | `EFCE` |
| this move:                                 |
| fa_avianex                | `F0E6` | `EFC2` | 
| these swap:                              |
| fa_diamond                | `F219` | `F29F` |
| fa_gem                    | `F29F` | `F219` |
| fa_cloudsmith             | `F16A` | `F167` |
| fa_youtube                | `F167` | `F16A` |

### :warning: Major change
* Reorder some FontAwesome glyphs #1596

### Bugs
* Remove Ubuntu 'fi' and 'fl' ligs #1597

### Improvements

* Add individual licenses to font database `fonts.json` for package maintainers #1599
* Add sha256 checksums to release (commit c2cd3cdf)

**Reply by Finii** [reply]
* Precreate changelog (see above)
* Adjust package.json
* Trigger Release workflow, manually
* wait for actions to complete and do not tough repo before
* Edit changelog.md  and also put onto gh-pages as new release
* Edit release draft comment with changelog
* Prepare casks https://github.com/ryanoasis/nerd-fonts/discussions/1074#discussioncomment-5770142
  *  Sync fork of https://github.com/Finii/homebrew-cask-fonts
  *  PAT should be still ok
* Publish Release
* Trigger Cask workflow manually
* Create PR is cask fork
* Trigger FontPatcher.zip workflow manually

**Reply by Finii** [reply]
Strangely the release workflow created 2 release drafts, and put one font (daddytimemono) in one and all other files in the other.
Manually copied the files over and deleted the stray draft-release.
:thinking:
